src/SFA.DAS.EmployerIncentives.Domain/Accounts/Account.cs

```diff
@@ -55,13 +55,27 @@ public void AddLegalEntity(long accountLegalEntityId, LegalEntity legalEntity)
         }
 
         public void UpdateVendorRegistrationCaseStatus(string hashedLegalEntityId, string caseId, string status, DateTime lastUpdatedDate)
+        {
+            if (HasLegalEntitiesWithIncompleteVrf(hashedLegalEntityId))
+            {
+                AddEvent(new VendorRegistrationCaseStatusUpdatedForLegalEntity(hashedLegalEntityId, caseId, status, lastUpdatedDate));
+            }
+        }
+
+        private bool HasLegalEntitiesWithIncompleteVrf(string hashedLegalEntityId)
+        {
+            return LegalEntities.Any(x => x.HashedLegalEntityId == hashedLegalEntityId &&
+                                          LegalEntity.VrfStatusIsCompleted(x.VrfCaseStatus));
+        }
+
+        public void SetVendorRegistrationCaseDetails(string hashedLegalEntityId, string caseId, string status, DateTime lastUpdatedDate)
```

Comment:
Can this method be merged with UpdateVendorRegistrationCaseStatus?  They have the same parameters?